src/MongoDB.Driver.Core/ReadPreferenceHedge.cs

```diff
@@ -0,0 +1,142 @@
+        /// </summary>
+        public static ReadPreferenceHedge ServerDefault => __serverDefault;
+        #endregion
+
+        // public methods
+        /// <inheritdoc/>
+        public abstract bool Equals(ReadPreferenceHedge other);
+
+        /// <summary>
+        /// Returns the BsonDocument representation of a hedge.
+        /// </summary>
+        /// <returns>A BsonDocument.</returns>
+        public abstract BsonDocument ToBsonDocument();
+
+        /// <inheritdoc/>
+        public override string ToString()
+        {
+            return ToBsonDocument().ToJson();
+        }
+    }
+
+    /// <summary>
+    /// Represents a custom read preference hedge.
+    /// </summary>
+    public sealed class CustomReadPreferenceHedge : ReadPreferenceHedge
+    {
+        // private fields
+        private readonly bool _isEnabled;
+
+        // constructors
+        /// <summary>
+        /// Initializes an instance of CustomReadPreferenceHedge.
+        /// </summary>
+        /// <param name="isEnabled">Whether hedged reads are enabled.</param>
+        public CustomReadPreferenceHedge(bool isEnabled)
+        {
+            _isEnabled = isEnabled;
+        }
+
+        // public properties
+        /// <summary>
+        /// Gets whether hedged reads are enabled.
+        /// </summary>
+        public bool IsEnabled => _isEnabled;
+        
+        // public methods
+        /// <inheritdoc/>
+        public override bool Equals(object other)
+        {
+            return Equals(other as CustomReadPreferenceHedge);
+        }
+
+        /// <inheritdoc/>
+        public override bool Equals(ReadPreferenceHedge other)
+        {
+            return
+                other is CustomReadPreferenceHedge customOther &&
+                _isEnabled == customOther._isEnabled;
+        }
+
+        /// <inheritdoc/>
+        public override int GetHashCode()
+        {
+            return _isEnabled.GetHashCode();
+        }
+
+        /// <inheritdoc/>
+        public override BsonDocument ToBsonDocument()
+        {
+            return new BsonDocument("enabled", _isEnabled);
+        }
+    }
+
+    /// <summary>
+    /// Represents the server default read preference hedge.
+    /// </summary>
+    public sealed class ServerDefaultReadPreferenceHedge : ReadPreferenceHedge
+    {
+        // public methods
+        /// <inheritdoc/>
+        public override bool Equals(object other)
+        {
+            return Equals(other as ServerDefaultReadPreferenceHedge);
+        }
+
+        /// <inheritdoc/>
+        public override bool Equals(ReadPreferenceHedge other)
+        {
+            return other is ServerDefaultReadPreferenceHedge;
+        }
+
+        /// <inheritdoc/>
+        public override int GetHashCode()
+        {
+            return 0;
+        }
+
+        /// <inheritdoc/>
+        public override BsonDocument ToBsonDocument()
+        {
+            return new BsonDocument();
```

Comment:
This appears to be out of line with the spec, which says:

> ... Any driver that exposes a builder API for read preference objects MUST NOT allow an empty ``hedge`` document to be constructed. In this case, the user MUST specify a value for ``enabled``, which MUST default to ``true``. If the user does not call a ``hedge`` API method, drivers MUST NOT send a ``hedge`` option to the server.

Reply:
I've removed the ability to specify a `{ hedge : { } }` version at your request.